plugins/modules/foreman_scap_content.py

```diff
@@ -130,6 +131,11 @@ def main():
             if not entity and 'scap_file' not in module_params:
                 module.fail_json(msg="Content of scap_file not provided. XML containing SCAP content is required.")
 
+            if entity and 'scap_file' in module_params:
+                digest = hashlib.sha256(module_params['scap_file'].encode("utf-8")).hexdigest()
```

Comment:
Does `module.sha256` not work here?

Reply:
It expects a path, this is not a path, see my comment :)

Reply:
Oh right. The notification took me directly to the diff...
Given that we'd probably don't want large amounts of xml embedded in yaml, I suspect passing it as a file seems to be the better UX.

Reply:
yeah, passing it as a file seems to be the better UX.

Reply:
For some reason I fail to calculate the same checksum as Foreman is when I do it as a file, damn -.-

Reply:
Is he stripping more Whitespace?

Reply:
Winner Winner, Chicken Dinner!

```diff
--- tests/test_playbooks/data/ssg-firefox-ds.xml	2020-03-25 13:55:54.952567896 +0100
+++ /home/evgeni/Downloads/ssg-firefox-ds.xml	2020-03-25 15:05:39.476260242 +0100
@@ -4012,4 +4012,4 @@
       </ns12:cpe-item>
     </ns12:cpe-list>
   </ns0:component>
-</ns0:data-stream-collection>
+</ns0:data-stream-collection>
\ No newline at end of file
```

Reply:
Sure It is better UX.
Looking at the foreman_openscap plugin itself, I see that it is using SHA2 digest.
https://github.com/theforeman/foreman_openscap/blob/master/lib/foreman_openscap/bulk_upload.rb#L24

Reply:
I must somehow miss your comment Evgeni. I guess you got it. :+1:

Reply:
Yeah, I think I got it working now.